style/values/computed/length_percentage.rs

```diff
@@ -530,8 +530,11 @@ impl LengthPercentage {
     /// Convert the computed value into used value.
     #[inline]
     pub fn maybe_to_used_value(&self, container_len: Option<Au>) -> Option<Au> {
-        self.maybe_percentage_relative_to(container_len.map(Length::from))
-            .map(Au::from)
+        if let Unpacked::Percentage(_) = self.unpack() {
+            return self.maybe_percentage_relative_to(container_len.map(Length::from))
+                .map(|length| Au::from_f32_px_trunc(length.px()));
+        }
+        self.maybe_percentage_relative_to(container_len.map(Length::from)).map(Au::from)
```

Comment:
No need to repeat `self.maybe_percentage_relative_to(container_len.map(Length::from))`

```suggestion
        self.maybe_percentage_relative_to(container_len.map(Length::from))
            .map(if let Unpacked::Percentage(_) = self.unpack() {
                |length| Au::from_f32_px_trunc(length.px())
            } else {
                Length::from
            })
    }
```

Reply:
fixed.